### DISCUSSION TITLE
Cards no Bootstrap 4

### DISCUSSION THREAD

**Discussion by joseliacosta** [frontendbr/forum#834] [Perguntas e Respostas]
_Disclaimer: eu sei que compensa muito mais fazer a maior parte das coisas na mão, mas meu time adotou o Bootstrap como framework então estou procurando a melhor forma de trabalhar com ele._

Vocês que estão utilizando o Bootstrap 4, tem conseguido usar o componente `Card` e tratar manualmente a largura dele com as colunas que o framework disponibiliza?
Eu percebi na documentação que agora todas as colunas ajustam a altura conforme o conteúdo. O que me parece ótimo. Porém se você está usando um componente dentro do grid, não necessariamente isso vai acontecer.
Então vi na parte da documentação dos [Cards](https://v4-alpha.getbootstrap.com/components/card/) que existe uma classe chamada `card-deck` que provê ao componente o que a coluna também provê: a mesma altura para todos, de acordo com o tamanho do maior deles. Mas o que tem me intrigado é que parece que eu abro mão de controlar a largura se eu quiser ter a altura 100%. E o mesmo acontece se eu escolho controlar a largura pelas classes `col-md-4`, por exemplo.
fiz um pequeno codepen para ilustrar essa dúvida

https://codepen.io/joselialcosta/pen/XzxpKy

Obrigada!

UPDATE: estou usando a versão v4-alpha

**Comment by klarkc** [comment]
Acho que se vc usar o flex em conjunto com as colunas deve resolver:

https://getbootstrap.com/docs/4.0/utilities/flex/

Exemplo: https://codepen.io/anon/pen/POXBEP

**Comment by joseliacosta** [comment]
Oi @klarkc eu esqueci de mencionar um detalhe: estou usando a versão alpha, ainda 😞 
Daí seguindo o seu pen eu modifiquei a versão e ele também quebra. 🎻 [sad music]

**Comment by klarkc** [comment]
Acho q nada impede de vc mesmo criar uma classe, chamada também `d-flex` que adiciona `align-items: stretch; display: flex;`. Como já está especificado na versão 4 final vc n vai ter problemas no futuro.

**Comment by joseliacosta** [comment]
Entendi @klarkc . O que me irrita um pouco é que, embora não seja algo dificil de contornar não venha como `default` essa possibilidade de customizar usando a propria documentação. Penso que o preço de usar Bootstrap é muito alto para a aplicação pra ele não oferecer determinadas coisas.
